test/elements/test_libraryelement.py

```diff
@@ -0,0 +1,1123 @@
+                        "updatedBy": {
+                            "avatarUrl": "/avatar/3dbccb5e89c37491dae50c6a4be200c6",
+                            "id": 2,
+                            "name": "sa-tibco-messaging-monitoring",
+                        },
+                    },
+                },
+            ],
+            "page": 1,
+            "perPage": 100,
+        }
+    }
+
+    HealthResponsePre8_2: dict = {"commit": "unknown", "database": "ok", "version": "8.1.8"}
+    HealthResponsePost8_2: dict = {"commit": "unknown-dev", "database": "ok", "version": "10.2.2"}
+
+    def setUp(self):
+        self.grafana = GrafanaApi(("admin", "admin"), host="localhost", url_path_prefix="", protocol="http")
+
+    @requests_mock.Mocker()
+    def test_get_library_element(self, m):
+        m.get(
+            "http://localhost/api/health",
+            json=LibraryElementTestCase.HealthResponsePost8_2,
+        )
+        m.get(
+            f"http://localhost/api/library-elements/{LibraryElementTestCase.ConnectedPanelUID}",
+            json=LibraryElementTestCase.ConnectedPanelJSON,
+        )
+        element = self.grafana.libraryelement.get_library_element(LibraryElementTestCase.ConnectedPanelUID)
+        self.assertEqual(element["uid"], LibraryElementTestCase.ConnectedPanelUID)
+
+    @requests_mock.Mocker()
+    def test_get_library_element_grafana_pre_8_2(self, m):
+        m.get(
+            "http://localhost/api/health",
+            json=LibraryElementTestCase.HealthResponsePre8_2,
+        )
+        m.get(
+            f"http://localhost/api/library-elements/{LibraryElementTestCase.ConnectedPanelUID}",
+            json=LibraryElementTestCase.ConnectedPanelJSON,
+        )
+        with self.assertRaises(DeprecationWarning) as ex:
+            self.grafana.libraryelement.get_library_element(LibraryElementTestCase.ConnectedPanelUID)
+        self.assertEqual(
+            "Grafana versions earlier than 8.2 do not support library elements",
+            str(ex.exception),
+        )
+
+    @requests_mock.Mocker()
+    def test_get_library_element_notfound(self, m):
+        def custom_matcher(request):
+            if request.path_url == "/api/health":
+                return None
+            resp = requests.Response()
+            resp.status_code = 404
+            resp._content = b'{"message": "library element could not be found"}'
+            resp.headers["Content-Type"] = "application/json"
+            return resp
+
+        m.get(
+            "http://localhost/api/health",
+            json=LibraryElementTestCase.HealthResponsePost8_2,
+        )
+        m.add_matcher(custom_matcher)
+        with self.assertRaises(GrafanaClientError) as ex:
+            self.grafana.libraryelement.get_library_element(LibraryElementTestCase.MissingPanelUID)
+        self.assertEqual(404, ex.exception.status_code)
+        self.assertEqual("Client Error 404: library element could not be found", ex.exception.message)
```

Comment:
I guess the reason for the hiccup is coming from this slightly different way to establish a request mock. I already tried to use `resp = niquests.Response()` here, but it still shows the same error.

```python
AttributeError: 'Response' object has no attribute '_promise'
```

Reply:
Immediately, yes, the mocking is the issue here.
The "raw" attribute must be set within the response, even if it's a empty BytesIO.

I am going to look deeper into it as soon as I can.

Reply:
I've replaced it by using the regular way of mocking. Might be an edge case to consider in order to improve interaction between niquests and requests_mock.

Reply:
OK. I've noted it on my side.

Reply:
The CI failure is unrelated to us. httpbin.org flakyness.

Reply:
Fixed with e1c4358.